### PR TITLE
25 password functionality

### PR DESCRIPTION
**Pull request by juliano60**
Password functionality implementation, first stab
@ZacTredger , @michowski Could you please review?

Kindest Regards,
Jules

### CODE REVIEW

File: src/main/java/newbank/server/Customer.java [hunk 1]
```diff
@@ -3,9 +3,15 @@
 import java.util.ArrayList;
 
 public class Customer {
-	
+	private String username;
+	private String password;
 	private ArrayList<Account> accounts;
 	
+	public Customer( final String username, final String password ) {
```
Comment (michowski) [inline src/main/java/newbank/server/Customer.java]: These parenthesis seem inconsistent with the rest of the code.

File: src/main/java/newbank/server/NewBank.java [hunk 2]
```diff
@@ -26,16 +28,44 @@ private void addTestData() {
...
+				return new CustomerID( username );	
+			}
 		}
 		return null;
 	}
+	
+	private boolean authenticateCustomer( final String username, final String password ) {
+		assert( customers.containsKey( username) );
+		
+		Optional<Map.Entry<String, Customer>> customer = customers.entrySet().stream()
```
Comment (michowski) [inline src/main/java/newbank/server/NewBank.java]: Isn't there a simpler one-liner to get that entry?

File: src/main/java/newbank/server/Customer.java [hunk 3]
```diff
@@ -3,9 +3,15 @@
...
+		this.username = username;
+		this.password = password;
+	}
+	
 	public Customer() {
```
Comment (michowski) [inline src/main/java/newbank/server/Customer.java]: I don't think we should support this overloaded constructor anymore. Every customer shall have a username and password.

**Review by michowski — COMMENTED** [review]
Hi Jules, great stuff!!! A few comments from my side.

File: src/main/java/newbank/server/Customer.java (hunk 1 above)
Comment (juliano60) [reply src/main/java/newbank/server/Customer.java]: Correct, fixed.

File: src/main/java/newbank/server/NewBank.java (hunk 2 above)
Comment (juliano60) [reply src/main/java/newbank/server/NewBank.java]: Correct. fixed

**Review by ZacTredger — APPROVED** [review]
This looks good to me after the changes @michowski suggested.
There's a bit of inherited awkwardness around `checkLogInDetails()` and `credentialsAreValid()`, but I'll try to address that this evening when I refactor the originator's stuff.

**Comment by juliano60** [comment]
Cheers guys.

Kindest Regards,
Jules

**Pull request merged**